### PR TITLE
Application Gateway deployment with Ssl-Offload

### PR DESCRIPTION
**Pull request by puneetsaraswat**
This 101 template is to demonstrate how to deploy an Application Gateway
with Ssl offload. You are required to provide a ssl certificate.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@puneetsaraswat__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (punsaras). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by kenazk** [comment]
Hi, could you rename the parameters file azuredeploy.parameters.json?

**Comment by puneetsaraswat** [comment]
The issue pointed out by acomghbot:
For certData and password parameters, we do not have any values to provide. These are customer owned certificates, we donot have a defaults for them.

File: 101-create-applicationgateway-publicip-ssl-offload/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Application Gateway With Public IP and HTTPS Listener
+
+Built by: [puneetsaraswat](https://github.com/puneetsaraswat)
```
Comment (singhkay) [inline 101-create-applicationgateway-publicip-ssl-offload/README.md]: Please remove this

File: 101-create-applicationgateway-publicip-ssl-offload/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "metadata": {
+                "description": "Location to deploy to"
+            }
+        },
+        "addressPrefix": {
```
Comment (singhkay) [inline 101-create-applicationgateway-publicip-ssl-offload/azuredeploy.json]: Does the user need to provide the address prefixes for a 101 template or can these become variables?

File: 101-create-applicationgateway-publicip-ssl-offload/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+            "type": "string",
+            "defaultValue": "10.0.0.0/16",
+            "metadata": {
+                "description": "Address prefix for the Virtual Network"
+            }
+        },
+        "subnetPrefix": {
```
Comment (singhkay) [inline 101-create-applicationgateway-publicip-ssl-offload/azuredeploy.json]: Does the user need to provide the address prefixes for a 101 template or can these become variables?

File: 101-create-applicationgateway-publicip-ssl-offload/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+        "publicIPAddressName": "publicIp1",
+        "virtualNetworkName": "virtualNetwork1",
+        "subnetName": "appGatewaySubnet",
+        "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+        "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
+        "publicIPRef": "[resourceId('Microsoft.Network/publicIPAddresses',variables('publicIPAddressName'))]",
+        "applicationGatewayID": "[resourceId('Microsoft.Network/applicationGateways',variables('applicationGatewayName'))]"
+    },
+    "resources": [{
+      "apiVersion": "2015-05-01-preview",
```
Comment (singhkay) [inline 101-create-applicationgateway-publicip-ssl-offload/azuredeploy.json]: Probably a good idea to make the api version a variable to move to newer versions quickly

**Comment by singhkay** [comment]
@puneetsaraswat The bot issue is that the certData and certPassword parameters are not included in the parameters file. 

Additionally, can we rename the public ip templates to 
101-application-gateway-public-ip
101-application-gateway-public-ip-ssl-offload

so that the name fits in the GIthub default folder column width and reads a little easier

**Pull request merged**